src/builder.rs

```diff
@@ -980,7 +980,9 @@ impl<'a, 'gcc, 'tcx> BuilderMethods<'a, 'tcx> for Builder<'a, 'gcc, 'tcx> {
 
     fn volatile_load(&mut self, ty: Type<'gcc>, ptr: RValue<'gcc>) -> RValue<'gcc> {
         let ptr = self.context.new_cast(self.location, ptr, ty.make_volatile().make_pointer());
-        ptr.dereference(self.location).to_rvalue()
+        let local = self.current_func().new_local(self.location, ty, "volatile_tmp");
```

Comment:
Please add a note like in `load()` to explain why we do this so that someone doesn't remove it at some point.